/icp_localization:
  ros__parameters:
    # pcd_file_path: "src/rm_simulation/pb_rm_simulation/meshes/RMUC2024_world/preview/RMUC.pcd"
    pcd_file_path: "src/rm_simulation/pb_rm_simulation/meshes/auto_world/preview/RMUL.pcd"
    icp_localization_ros2:
      range_data_topic: "/cloud_registered_body"
      num_accumulated_range_data: 1
      imu_data_topic: "/imu/data"
      odometry_data_topic: "/Odometry" # Utilize '/Odometry' from FAST_LIO when in REAL environment
      is_use_odometry: true
      is_provide_odom_frame: true
      gravity_vector_filter_time_constant: 0.01
      fixed_frame: "map"
      min_num_odom_msgs_before_ready: 10

    initial_pose:
      x: 0.0
      y: 0.0 
      z: 0.0 
      roll: 0.0  # degrees  
      pitch: 0.0 # degrees
      yaw: 0.0  # degrees

    calibration:
      imu_to_range_sensor:
        x: 0.0
        y: 0.0
        z: -0.025
        roll: 0.0
        pitch: 0.0
        yaw: 0.0

      odometry_source_to_range_sensor:
        x: 0.0
        y: 0.0
        z: 0.0
        roll: 0.0
        pitch: 0.0
        yaw: 0.0 
